pkg/api/metrics.go

```diff
@@ -60,10 +62,14 @@ func (hs *HTTPServer) QueryMetricsV2(c *contextmodel.ReqContext) response.Respon
 	if err != nil {
 		return hs.handleQueryMetricsError(err)
 	}
-	return hs.toJsonStreamingResponse(resp)
+	return hs.toJsonStreamingResponse(c.Req.Context(), resp)
 }
 
-func (hs *HTTPServer) toJsonStreamingResponse(qdr *backend.QueryDataResponse) response.Response {
+func (hs *HTTPServer) toJsonStreamingResponse(ctx context.Context, qdr *backend.QueryDataResponse) response.Response {
+	// treat response as downstream since it means we've got
+	// a proper response from the plugin.
+	requestmeta.WithDownstreamStatusSource(ctx)
```

Comment:
Please refer to the PR description of how it currently works and reason why. 

> if the error is raised before we proxy the request it should be server. if the requests returns an proxied response it should be downstream.

We got a `backend.QueryDataResponse` so response is downstream. Otherwise server by default. What am I missing?

Reply:
sorry... typo :(

If the error is raised before we proxy the request it should be server. 
if the requests returns an proxied error response it should be downstream.
If the request returns an proxied successful response it should be server.

the label should only change if the downstream response is an error. This way IF there is something that breaks after the datasource request it would be treated as an error in the server and count as an failed request. errors from downstream systems should not count as failed requests.

Reply:
> If the request returns an proxied successful response it should be server.

This is the last thing to discuss then I guess since this currently returns downstream.

Reply:
If error or response includes an error set downstream